internal/events/dx_callbacks.go

```diff
@@ -117,30 +117,32 @@ func (em *eventManager) checkReceivedIdentity(peerID string, author string) (nod
 }
 
 func (em *eventManager) pinedBatchReceived(peerID string, batch *fftypes.Batch) error {
-	l := log.L(em.ctx)
 
 	// Retry for persistence errors (not validation errors)
 	return em.retry.Do(em.ctx, "private batch received", func(attempt int) (bool, error) {
+		return true, em.database.RunAsGroup(em.ctx, func(ctx context.Context) error {
+			l := log.L(ctx)
 
-		node, err := em.checkReceivedIdentity(peerID, batch.Author)
-		if err != nil {
-			return true, err
-		}
-		if node == nil {
-			l.Errorf("Batch received from invalid author '%s' for peer ID '%s'", batch.Author, peerID)
-			return false, nil
-		}
+			node, err := em.checkReceivedIdentity(ctx, peerID, batch.Author)
+			if err != nil {
+				return err
+			}
+			if node == nil {
+				l.Errorf("Batch received from invalid author '%s' for peer ID '%s'", batch.Author, peerID)
+				return nil
```

Comment:
In the original version, this was returning `false, nil`. It looks like now we return `true, nil`. does that change any behavior?

Reply:
Great comment. The retry logic ignores `true` in the case of an `err == nil`:
https://github.com/hyperledger-labs/firefly/blob/a49e6e63345bfcb5cd30e1c33e99b38b844e2294/internal/retry/retry.go#L59-L61